packages/runtime/runtime-utils/src/requestParser.ts

```diff
@@ -17,21 +17,25 @@ export class RequestParser implements IRequest {
 	 */
 	public static getPathParts(url: string): readonly string[] {
 		const queryStartIndex = url.indexOf("?");
-		return url
-			.substring(0, queryStartIndex < 0 ? url.length : queryStartIndex)
-			.split("/")
-			.reduce<string[]>((pv, cv) => {
-				if (cv !== undefined && cv.length > 0) {
-					pv.push(decodeURIComponent(cv));
-				}
-				return pv;
-			}, []);
+		const pathParts: string[] = [];
+		const urlPath = url.slice(
+			0,
+			Math.max(0, queryStartIndex < 0 ? url.length : queryStartIndex),
```

Comment:
I think the `Math.max()` here is unnecessary? I get it's so `slice()` doesn't get a negative argument in case `queryStartIndex === -1`, but in this case the ternary already handles that.
```suggestion
			queryStartIndex < 0 ? url.length : queryStartIndex,
```